nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/elasticsearch/ElasticsearchObject.java

```diff
@@ -0,0 +1,131 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.rest.management.elasticsearch;
+
+import static javax.servlet.http.HttpServletResponse.SC_NOT_FOUND;
+import static javax.servlet.http.HttpServletResponse.SC_OK;
+import static org.nuxeo.ecm.core.api.security.SecurityConstants.SYSTEM_USERNAME;
+import static org.nuxeo.elasticsearch.bulk.IndexAction.ACTION_NAME;
+import static org.nuxeo.elasticsearch.bulk.IndexAction.INDEX_UPDATE_ALIAS_PARAM;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+
+import org.apache.commons.lang3.StringUtils;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.bulk.BulkService;
+import org.nuxeo.ecm.core.bulk.message.BulkCommand;
+import org.nuxeo.ecm.core.bulk.message.BulkStatus;
+import org.nuxeo.ecm.core.query.sql.NXQL;
+import org.nuxeo.ecm.core.repository.RepositoryService;
+import org.nuxeo.ecm.webengine.model.WebObject;
+import org.nuxeo.ecm.webengine.model.impl.DefaultObject;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Endpoint to administer Elasticsearch
+ *
+ * @since 11.1
+ */
+@WebObject(type = "elasticsearch")
+public class ElasticsearchObject extends DefaultObject {
+
+    /**
+     * Performs an ES indexing on documents of the given repository and that match the NXQL query.
+     *
+     * @see #performIndexing(String, String)
+     */
+    @POST
+    @Path("{repositoryName}/reindex")
+    public Response doIndexing(@PathParam("repositoryName") String repositoryName, @QueryParam("query") String query) {
+        return performIndexing(repositoryName, query);
+    }
+
+    /**
+     * Performs an ES indexing on the given document and its children for a given repository.
+     *
+     * @see #performIndexing(String, String)
+     */
+    @POST
+    @Path("{repositoryName}/{documentId}/reindex")
+    public Response doIndexingOnDocument(@PathParam("repositoryName") String repositoryName,
+            @PathParam("documentId") String documentId) {
+        String query = String.format("Select * From Document where %s = '%s' or %s = '%s'", //
+                NXQL.ECM_UUID, documentId, //
+                NXQL.ECM_PARENTID, documentId);
```

Comment:
We want it recursive here, use `ancestorId`.

Reply:
yep, i had a doubt about it, hence my question 👍